candidate_models/model_commitments/vonecornets.py

```diff
@@ -0,0 +1,93 @@
+
+from candidate_models.model_commitments.cornets import CORnetCommitment, _build_time_mappings, find_nearest
+import logging
+
+import numpy as np
+from tqdm import tqdm
+
+from brainio_base.assemblies import merge_data_arrays, NeuroidAssembly, walk_coords
+from candidate_models.base_models import vonecornet
+from result_caching import store
+from brainscore.submission.utils import UniqueKeyDict
+from brainscore.utils import LazyLoad
+_logger = logging.getLogger(__name__)
+
+
+class VOneCORnetCommitment(CORnetCommitment):
+    def start_recording(self, recording_target, time_bins):
+        if recording_target == 'V1':
+            self.recording_layers = ['vone_block.output-t0']
+        else:
+            self.recording_layers = [layer for layer in self.layers if recording_target in layer]
+        self.recording_time_bins = time_bins
+
+    @store(identifier_ignore=['stimuli'])
+    def look_at_cached(self, model_identifier, stimuli_identifier, stimuli):
```

Comment:
this looks like a lot of duplicate code, is there a way to only separate the differences?

Reply:
The only reason that I had to replicate the look_at_cached function was to add lines 33 to 36. Currently in the activations, the 'region' attribute is obtained by the layer name and not by the area that it is recording from. This is because in the CORnet architectures the layers are named V1.xxx, V2.xxx, etc... In the VOneCORnet that is not the case. and so inside the activation regions the attribute is vone_block, model.V2, model.V4, etc... 

A way around it would be to change the CORnetCommitment instead of getting the regions from the responses['region'] it would store the recording region in the start_recording method. What do you think about this?

Something like:

```
    def start_recording(self, recording_target, time_bins):
        self.recording_target = recording_target
        self.recording_layers = [layer for layer in self.layers if layer.startswith(recording_target)]
        self.recording_time_bins = time_bins
```


```
    @store(identifier_ignore=['stimuli'])
    def look_at_cached(self, model_identifier, stimuli_identifier, stimuli):
        responses = self.activations_model(stimuli, layers=self.recording_layers)
        # map time
        regions = self.recording_target

```

This way the VOneCORnetCommitment would not need to add those lines to change the regions

Reply:
your suggestion makes sense to me. Alternatively, it's probably also fine to leave it as-is, Jonas at some point convincingly argued for more independence of models. I'll leave it up to you to decide what makes most sense